rosa_hcp/terraform/rosa-hcp-creating-a-cluster-with-customizations-terraform.adoc

```diff
@@ -0,0 +1,72 @@
+:_mod-docs-content-type: ASSEMBLY
+[id="rosa-sts-creating-a-cluster-with-customizations-terraform-hcp"]
+= Customizing a ROSA cluster with Terraform
+include::_attributes/attributes-openshift-dedicated.adoc[]
+:context: rosa-sts-creating-a-cluster-with-customizations-terraform-hcp
+
+toc::[]
+
+You can use Terraform to create a customized {product-title} (ROSA) cluster by using Terraform to customize the cluster's details.
+
+[NOTE]
+====
+This guide assumes a basic understanding of ROSA. See the additional resources for conceptual information about ROSA clusters.
+====
+
+.Prerequisites
+
+* You have generated and stored an link:https://console.redhat.com/openshift/token[offline {cluster-manager-first} token].
+* You have an AWS account with the correct permissions.
+* You completed the xref:../../rosa_planning/rosa-cloud-expert-prereq-checklist.adoc[Prerequisites checklist for deploying ROSA using STS]
+* You completed the setup steps in xref:../../rosa_planning/rosa-understanding-terraform.adoc[Preparing Terraform to install ROSA clusters]
+* You have installed link:https://developer.hashicorp.com/terraform/install[the `terraform` CLI tool].
+
+[id="rosa-sts-creating-a-cluster-with-customizations-terraform-creating-hcp"]
+== Creating a ROSA Classic cluster using Terraform
+
+The cluster creation process outlined below shows how to use Terraform to create your account-wide IAM roles and a default ROSA cluster with a managed OIDC configuration.
+
+[NOTE]
+====
+You edit the default `main.tf`, `variables.tf`, and `terraform.tfvars` files to customize a cluster. Cluster customizations must be performed before you create a cluster. To customize, copy the default Terraform file from the procedure below and make the desired changes.
+====
+
+include::modules/rosa-sts-cluster-terraform-setup.adoc[leveloffset=+2]
+include::modules/rosa-hcp-cluster-terraform-file-creation.adoc[leveloffset=+2]
+
+[id="rosa-sts-creating-a-cluster-with-customizations-terraform-customizing-hcp"]
+== Terraform customization options
+
+The following sections detail individual customizations you can add to your `main.tf`, `variables.tf`, and `terraform.tfvars` files.
+
+include::modules/rosa-cluster-cluster-role-name-change.adoc[leveloffset=+2]
+include::modules/rosa-cluster-enable-autoscaling-terraform.adoc[leveloffset=+2]
+include::modules/rosa-sts-cluster-terraform-execute.adoc[leveloffset=+1]
+include::modules/rosa-sts-cluster-terraform-destroy.adoc[leveloffset=+1]
+
+[id="next-steps_{context}"]
+== Next steps
+
+* xref:../../rosa_install_access_delete_clusters/rosa-sts-accessing-cluster.adoc#rosa-sts-accessing-cluster[Accessing a ROSA cluster]
+
+[role="_additional-resources"]
+[id="additional-resources_rosa-sts-creating-a-cluster-with-customizations-terraform"]
+== Additional resources
+
```

Comment:
This is a lot of additional resources. Is there a way to break them into categories or something for easier reading?